File: sdk/runanywhere-flutter/packages/runanywhere/lib/capabilities/model_loading/models/loaded_model.dart
import 'package:runanywhere/core/models/common.dart';
import 'package:runanywhere/core/module_registry.dart';

/// Represents a model that has been loaded and is ready for use
class LoadedModel {
  /// The model information
  final ModelInfo model;

  /// The service that can execute this model
  final LLMService service;

  LoadedModel({
    required this.model,
    required this.service,
  });
}

File: sdk/runanywhere-react-native/src/Capabilities/ModelLoading/Models/LoadedModel.ts
/**
 * LoadedModel.ts
 *
 * Represents a model that has been loaded and is ready for use
 *
 * Reference: sdk/runanywhere-swift/Sources/RunAnywhere/Capabilities/ModelLoading/Models/LoadedModel.swift
 */

import type { ModelInfo } from '../../../Core/Models/Model/ModelInfo';
import type { LLMService } from '../../../Core/Protocols/LLM/LLMService';

/**
 * Represents a model that has been loaded and is ready for use
 */
export interface LoadedModel {
  /** The model information */
  readonly model: ModelInfo;

  /** The service that can execute this model */
  readonly service: LLMService;
}

/**
 * Create loaded model
 */
export class LoadedModelImpl implements LoadedModel {
  public readonly model: ModelInfo;
  public readonly service: LLMService;

  constructor(model: ModelInfo, service: LLMService) {
    this.model = model;
    this.service = service;
  }
}
